### ISSUE TITLE
@Bind (FieldSerializerAnnotationsUtil) does not pass field type to serializer constructor 

### ISSUE THREAD

**Issue by tweise** [EsotericSoftware/kryo#310]
Instead of field.getClass() it should be field.getType() in

https://github.com/EsotericSoftware/kryo/blob/master/src/com/esotericsoftware/kryo/serializers/FieldSerializerAnnotationsUtil.java#L69

```
Serializer s = ReflectionSerializerFactory.makeSerializer(fieldSerializer.getKryo(), serializerClass, field.getClass());
```

**Comment by magro** [comment]
Sorry for the late response, is this still relevant? If so, can you elaborate the issue?

**Comment by magro** [comment]
Please reopen if still relevant.

**Comment by tweise** [comment]
@magro The bug is still there. Instead of passing the type of field, java.lang.reflect.Field is passed to makeSerializer (which expects the type). I don't see a way to reopen the issue, can you please do it.

**Comment by magro** [comment]
Ok. Do you want to submit a pull request?

**Comment by magro** [comment]
@tweise can you tell what the consequence of the issue is? How could it be reproduced?

**Comment by baltiyskiy** [comment]
I've just ran into exactly the same issue. The consequence is that FieldSerializer is instantiated for type `Field` instead of the proper element type, so the elements don't get serialized. That's pretty broken.

**Comment by baltiyskiy** [comment]
The tests didn’t catch this because BindCollection was only tested with LongArraySerializer, which doesn’t accept element class in the constructor. Actually, there should be more tests for CollectionBind and other binds with different serializers, but I think it’s out of scope for this issue.

**Comment by magro** [comment]
@NathanSweet Is this fixed in the 5.0 branch already?

**Comment by NathanSweet** [comment]
It should be now!
